src/Farmer/Farmer.Resources.fs

```diff
@@ -0,0 +1,882 @@
+// Contains the list of all ARM Azure resources supported by Farmer.
```

Comment:
We can split them up - there's absolutely no dependency across the resources contained here in terms of F# types. I see this as a growing "library" of supported ARM resources - whether we split them up by namespace e.g. Web, DocumentDb, EventHub etc. I don't mind.

The Resource Builders should be able to compose these resources together into what I'm starting to think of as "topologies" e.g. the web app builder combined the AI, ServerFarm and Website resources into a single logical entity. But we also have separate builders for e.g. AI in case you need to build your own one.

Moving to allowing us to create resources quickly through functions (rather than only via builders) adds some interesting possibilities for the future in terms of composition and reuse.

Reply:
Hmmm. If there are no dependence then the config and the builder together makes more sense to me than all the confines together. Put things together that change together.

Reply:
@dburriss builders / config etc. go together, yes. The underlying ARM resources I think should be separate because there won't always be a 1:1 between builders and the ARM resources themselves.

Reply:
I can understand not needing a builder for certain things. In the current incarnation, the Container registry hardly warrants a builder.

There are many pluses on this change but the biggest one for me is unlocking people to add ad-hoc resources as you showed above. Most of what I have been talking about here is just about files.
I am excited to see this change lands (and then update the contribution tutorial). It should simplify that a bit.

Nice one!